src/Forms/SearchableDropdownTrait.php

```diff
@@ -0,0 +1,594 @@
+<?php
+
+namespace SilverStripe\Forms;
+
+use ArrayAccess;
+use Error;
+use InvalidArgumentException;
+use LogicException;
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\ORM\DataList;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\DataObjectInterface;
+use SilverStripe\ORM\Relation;
+use SilverStripe\ORM\SS_List;
+use SilverStripe\ORM\FieldType\DBHTMLText;
+use SilverStripe\View\ArrayData;
+use SilverStripe\ORM\Search\SearchContext;
+use SilverStripe\Security\SecurityToken;
+use Exception;
+
+trait SearchableDropdownTrait
+{
+    private static array $allowed_actions = [
+        'search',
+    ];
+
+    private bool $isClearable = false;
+
+    private bool $isLazyLoaded = false;
```

Comment:
Lazy loading has a worse UX experience because of the delay of needing to make an AJAX request.  Lazy loading makes sense for large amounts of options, though for a small list of options it's better to not lazy load

Reply:
Ahhh right yup fair enough. In my head I was combining "search" and "lazyload" together for some reason - but yeah makes sense to load the initial values by default.